daemon/cmd/daemon_main.go

```diff
@@ -1642,6 +1649,33 @@ func initKubeProxyReplacementOptions() {
 			option.Config.NodePortMode = option.NodePortModeSNAT
 		}
 	}
+
+	if option.Config.EnableSessionAffinity {
+		if !probesManager.GetMapTypes().HaveLruHashMapType {
+			msg := "SessionAffinity feature requires BPF LRU maps"
+			if strict {
+				log.Fatal(msg)
+			} else {
+				log.Warn(fmt.Sprintf("%s. Disabling the feature.", msg))
+				option.Config.EnableSessionAffinity = false
+			}
+
+		}
+	}
+
+	if option.Config.EnableSessionAffinity && option.Config.EnableHostReachableServices {
+		found1, found2 := false, false
+		if h := probesManager.GetHelpers("cgroup_sock"); h != nil {
+			_, found1 = h["bpf_get_netns_cookie"]
+		}
+		if h := probesManager.GetHelpers("cgroup_sock_addr"); h != nil {
+			_, found2 = h["bpf_get_netns_cookie"]
+		}
+		if !(found1 && found2) {
+			log.Warnf("sessionAffinity for host reachable services needs kernel 5.7.0 or newer. " +
+				"Disabling sessionAffinity for cases when a service is accessed from a cluster.")
```

Comment:
Do you want to disable something here or is this message just for the user?

Reply:
Just a note for a user.

Reply:
Ahh I see bacause it's ifdef'ed with the netns helper define in bpf_sock, got it.